### ISSUE TITLE
[Bug]: show table result is not correct in no-sys account 

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#7897]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):d6e9798b8068f5b6358e7a26af1ef6d3b2e32f1a
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> create table f1(b int, a int primary key);
Query OK, 0 rows affected (0.01 sec)

mysql> create table f2(b int, aa varchar primary key);
Query OK, 0 rows affected (0.03 sec)

mysql> create table c1 (a int, b int, foreign key f_a(a) references f1(c));
ERROR 20101 (HY000): internal error: column 'c' no exists in table 'f1'
mysql> create table c1 (a int, b int, foreign key f_a(a) references f1(b));
ERROR 20101 (HY000): internal error: reference column 'b' is not unique constraint(Unique index or Primary Key)
mysql> create table c1 (a int, b int, foreign key f_a(a) references f2(aa));
ERROR 20101 (HY000): internal error: type of reference column 'aa' is not match for column 'a'
mysql> create table c1 (a int, b int, foreign key f_a(a) references f1(a));
Query OK, 0 rows affected (0.01 sec)

mysql> select * from f1;
Empty set (0.03 sec)

mysql> select * from c1;
Empty set (0.00 sec)

mysql> drop table f1;
ERROR 20101 (HY000): internal error: can not drop table 'f1' referenced by some foreign key constraint
mysql> truncate f1;
ERROR 20101 (HY000): internal error: can not truncate table 'f1' referenced by some foreign key constraint
mysql> truncate c1;
ERROR 20101 (HY000): internal error: can not find table by id 316775
mysql> drop table f1;
ERROR 20101 (HY000): internal error: can not drop table 'f1' referenced by some foreign key constraint
mysql> drop table c1;
ERROR 20101 (HY000): internal error: can not find table by id 316775
mysql> drop table f1;
ERROR 20101 (HY000): internal error: can not drop table 'f1' referenced by some foreign key constraint
mysql> drop table f2;
Query OK, 0 rows affected (0.01 sec)

**mysql> show tables;
+-------------------+
| Tables_in_fk_base |
+-------------------+
| f1                |
| c1                |
| t_dept            |
| t_emp             |
| t_dept1           |
| t_emp1            |
| t_dept2           |
| t_emp2            |
| f1                |
| c1                |
| t_dept            |
| t_emp             |
| t_dept1           |
| t_emp1            |
| t_dept2           |
| t_emp2            |
| f1                |
| c1                |
| f1                |
| c1                |
| m1                |
| f1                |
| c1                |
| f1                |
| c1                |
+-------------------+**
25 rows in set (0.14 sec)

### Steps to Reproduce

```Markdown
create table f1(b int, a int primary key);
create table f2(b int, aa varchar primary key);
create table c1 (a int, b int, foreign key f_a(a) references f1(c));
create table c1 (a int, b int, foreign key f_a(a) references f1(b));
create table c1 (a int, b int, foreign key f_a(a) references f2(aa));
create table c1 (a int, b int, foreign key f_a(a) references f1(a));
select * from f1;
select * from c1;
drop table f1;
truncate f1;
truncate c1;
```

**Comment by daviszhen** [comment]
I have reproduced partial phenomenon:

Create a general account `test` with admin_name `test`.

login with test:test.

mysql -h 127.0.0.1 -P 6001 -utest:test -p111 

create database fk_base;
use fk_base;

re-execute the statements.

I get the following results : 
```
mysql> show tables;
+-------------------+
| Tables_in_fk_base |
+-------------------+
| f1                |
| c1                |
| f1                |
| f2                |
| c1                |
+-------------------+
```
I try the select information from the mo_tables directly.
I get :

```
mysql> SELECT account_id,relname, relkind,rel_createsql FROM mo_catalog.mo_tables WHERE reldatabase='fk_base' ;
+------------+-------------------------+---------+---------------------------------------------------------------------+
| account_id | relname                 | relkind | rel_createsql                                                       |
+------------+-------------------------+---------+---------------------------------------------------------------------+
|       3001 | %!%mo_increment_columns |         |                                                                     |
|       3001 | f1                      | r       | create table f1(b int, a int primary key)                           |
|       3001 | c1                      | r       | create table c1 (a int, b int, foreign key f_a(a) references f1(a)) |
|       3001 | %!%mo_increment_columns |         |                                                                     |
|       3001 | f1                      | r       | create table f1(b int, a int primary key)                           |
|       3001 | f2                      | r       | create table f2(b int, aa varchar primary key)                      |
|       3001 | c1                      | r       | create table c1 (a int, b int, foreign key f_a(a) references f1(a)) |
+------------+-------------------------+---------+---------------------------------------------------------------------+
```

**Comment by daviszhen** [comment]
in progress

**Comment by daviszhen** [comment]
It was fixed at pr #7914

**Comment by aressu1985** [comment]
fixed: 4a2132a54f9e643346878e0be991c47dce965d9b
